tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,403 @@
+    # barrier_create('rke2_server_start_ready', 2);
+    barrier_wait('rke2_server_start_ready');
+
+    assert_script_run('mkdir -p ~/.kube');
+    assert_script_run('cp /etc/rancher/rke2/rke2.yaml ~/.kube/config');
+    assert_script_run('kubectl get nodes');
+
+    # Create registries ready
+    script_run('cat > /etc/rancher/rke2/registries.yaml <<__END
+mirrors:
+  registry.suse.de:
+    endpoint:
+      - http://registry.suse.com
+configs:
+  registry.suse.com:
+    tls:
+      insecure_skip_verify: true
+__END
+true');
+
+    # Wait for rke2-agent service to be ready
+    my $children = get_children();
+    mutex_wait('RKE2_AGENT_START_READY', (keys %$children)[0]);
+
+    assert_script_run("scp /etc/rancher/rke2/registries.yaml root\@$agent_ip:/etc/rancher/rke2/registries.yaml");
+
+    # Restart RKE2 service and check the service is active well after restart
+    systemctl('restart rke2-server.service', timeout => 180);
+    check_service_status();
+
+    # Restart rke2-server service complete
+    # barrier_create('rke2_server_restart_complete', 2);
+    barrier_wait('rke2_server_restart_complete');
+
+    mutex_wait('RKE2_AGENT_RESTART_COMPLETE', (keys %$children)[0]);
+
+    script_retry('! kubectl get nodes | grep NotReady', retry => 8, delay => 20, timeout => 180);
+    assert_script_run('kubectl get nodes');
+}
+
+sub check_service_status {
+    # Check RKE2 service status and error message
+    record_info('Check RKE2 service status and error message', '');
+    # systemctl('is-active rke2-server.service');
+    assert_script_run("systemctl status rke2-server.service | grep 'active (running)'");
+    assert_script_run("journalctl -u rke2-server | grep 'Started Rancher Kubernetes Engine v2 (server)'");
+    assert_script_run("! journalctl -u rke2-server | grep \'\"level\":\"error\"\'");
+}
+
+sub deploy_kubevirt_manifests {
+    # Install and deploy required kubevirt packages and manifests
+    my $os_version          = get_required_var('VERSION');
+    my $virt_tests_repo     = get_required_var('VIRT_TESTS_REPO');
+    my $virt_manifests_repo = get_var('VIRT_MANIFESTS_REPO');
+
+    record_info('Add kubevirt repo and install packages', '');
+    # Development Tools repo for OBS Module, e.g. http://download.suse.de/download/ibs/SUSE/Products/SLE-Module-Development-Tools-OBS/15-SP4/x86_64/product/
+    # Development product test repo for SLE official product OSD testing, e.g. http://download.suse.de/ibs/SUSE:/SLE-15-SP4:/GA/standard/
+    # Devel test repo, e.g. http://download.suse.de/download/ibs/Devel:/Virt:/SLE-15-SP4/SUSE_SLE-15-SP4_Update_standard/
+    # MU product test (SLE official MU channel+incidents)
+    zypper_call("ar $virt_tests_repo Virt-Tests-Repo");
+    zypper_call("ar $virt_manifests_repo Virt-Manifests-Repo") if ($virt_manifests_repo);
+    zypper_call("--gpg-auto-import-keys ref");
+
+    my $virt_manifests = 'containerized-data-importer-manifests kubevirt-manifests kubevirt-virtctl';
```

Comment:
Well, kubevirt-virtctl is just a rpm, not kubevirt manifest... Is it a little misleading with the function name?

Reply:
But it can be installed from the same repo 'Virt-Manifests-Repo'. How should we install it?

Reply:
It is fine to install it this way. How about modify the function name a little bit to be aligned with what this func really do?

Reply:
Well, I am not sure I get your logic for classifying this way... But to me, it is clear that containerized-data-importer-manifests kubevirt-manifests are manifest packages, while kubevirt-virtctl is a kubevirt command line tool pkg.

It seems you do not quite get my logic either ;-) Let me explain more. My point was about the function's name "deploy_kubevirt_manifests". The name shows that this function does manifest deployment, but actually it does more, which includes also kubevirt packages installation. So to me, it will be better to rename the function to eg "install_kubevirt_pkgs_and_deploy". HDYT?

Reply:
Yes, but the final purpose of this function is to deploy the manifest whatever it includes any presetup steps such as install related rpm packages before the manifest deployment. This is what it's the main thing to do. Besides, I don't want the function name too long ;), so I think to keep the current name is better and meaningful.

Reply:
I respect your decision :). 
It is not big deal anyway, but still I'd suggest to consider that the purpose of the author, and how the function name should be given are two things and have different rules, right?  Of course different rules can apply and fit in different scenario.

Reply:
Make sense. I changed the name to "install_kubevirt_pkgs_n_deploy_manifests" :). Thank you for the advice!